acnportal/algorithms/sorted_algorithms.py

```diff
@@ -21,166 +29,353 @@ class SortedSchedulingAlgo(BaseAlgorithm):
     This process continues until the list of EVs is empty.
 
-    The argument sort_fn controlled how the EVs are sorted. This controls which EVs will get potential higher charging
-    rates when infrastructure constrains become binding.
+    The argument sort_fn controlled how the EVs are sorted. This controls
+    which  EVs will get potential higher charging rates when infrastructure
+    constrains become binding.
 
     Args:
-        sort_fn (Callable[List[EV]]): Function which takes in a list of EVs and returns a list of the same EVs but
-            sorted according to some metric.
+        sort_fn (Callable[List[EV]]): Function which takes in a list of EVs
+            and returns a list of the same EVs but sorted according to some
+            metric.
+        continuous_inc (float): Increment to use when pilot signal is
+            continuously controllable.
     """
 
-    def schedule(self, active_evs):
+    def __init__(
+        self,
+        sort_fn,
+        estimate_max_rate=False,
+        max_rate_estimator=None,
+        uninterrupted_charging=False,
+        continuous_inc=0.1,
+        allow_overcharging=False,
+    ):
+        super().__init__(
+            sort_fn, estimate_max_rate, max_rate_estimator, uninterrupted_charging
+        )
+        self.continuous_inc = continuous_inc
+
+    def round_robin(self, active_sessions, infrastructure):
         """ Schedule EVs using a round robin based equal sharing scheme.
 
         Implements abstract method schedule from BaseAlgorithm.
 
         See class documentation for description of the algorithm.
 
         Args:
-            active_evs (List[EV]): see BaseAlgorithm
+            active_sessions (List[EV]): see BaseAlgorithm
+            infrastructure (InfrastructureInfo): Description of electrical
+                infrastructure.
 
         Returns:
             Dict[str, List[float]]: see BaseAlgorithm
         """
-        continuous_inc = 1
-
-        ev_queue = deque(self._sort_fn(active_evs, self.interface))
-        schedule = {ev.station_id: [0] for ev in active_evs}
-        rate_idx_map = {ev.station_id: 0 for ev in active_evs}
-        allowable_rates = {}
-        for ev in ev_queue:
-            evse_continuous, evse_rates = self.interface.allowable_pilot_signals(
-                ev.station_id
+        queue = deque(self._sort_fn(active_sessions, self.interface))
+        schedule = np.zeros(infrastructure.num_stations)
+        rate_idx = np.zeros(infrastructure.num_stations, dtype=int)
+        allowable_pilots = infrastructure.allowable_pilots.copy()
+        for session in queue:
+            i = infrastructure.get_station_index(session.station_id)
+            # If pilot signal is continuous discretize it with increments of
+            # continuous_inc.
+            if infrastructure.is_continuous[i]:
+                allowable_pilots[i] = np.arange(
+                    session.min_rates[0],
+                    session.max_rates[0] + 1e-7,
```

Comment:
in case for some reason we have an increment less than 1e-7, should we make the positive offset here tied to continuous_inc? Not saying this case is likely, but it might be better practice